src/components/File/index.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { PureComponent } from 'react';
+import { connect } from 'react-redux';
+import { createSelector } from 'reselect';
+import { format } from 'url';
+
+import { askNotificationPermission } from 'utils/browser-notifications';
+import { downloadSelector } from 'reducers/download';
+import { downloadURL } from 'actions/creators';
+
+import descriptionToPath from 'utils/processDescription/descriptionToPath';
+import blockEvent from 'utils/block-event';
+
+import FileButton, { SupportedExtensions } from './FileButton';
+
+const mapStateToPropsFor = (url: string, fileType: string, subset: boolean) =>
+  createSelector(
+    downloadSelector,
+    (downloads) =>
+      downloads[
+        [url, fileType, subset && 'subset'].filter(Boolean).join('|')
+      ] || {}
+  );
+
+type Props = {
+  api: ParsedURLServer;
+  entryDescription: Object;
+  customLocationDescription?: Object;
+  downloadURL: typeof downloadURL;
+  fileType: SupportedExtensions;
+  count: number;
+  subset?: boolean;
+  name: string;
+  search?: Record<string, string>;
+  endpoint?: string;
+  className?: string;
+  minWidth?: number | string;
```

Comment:
Allowing custom label:

```suggestion
  minWidth?: number | string;
  label?: string;
```